### DISCUSSION TITLE
`write_scrollback_file:open` do not work in i3 with xdg set to nvim

### DISCUSSION THREAD

**Discussion by edwin0cheng** [ghostty-org/ghostty#3585] [Feature Requests, Ideas] [answered]
I have the same problem (https://github.com/ghostty-org/ghostty/issues/2435#issuecomment-2425202971)  using i3 key binding to open ghostty.  

My xdg mimetype for `text/plain` is set to nvim, and when I use `write_scrollback_file:open`, it do launch a process in background (I checked by `ps aux | grep "nvim"`) , however it does not show up in desktop.

And I checked the `xdg-open` source code (which is a bash script, TIL), it does not respect the `TERMINAL` key described in https://specifications.freedesktop.org/desktop-entry-spec/latest/recognized-keys.html.  It just run the `EXEC` .

So in my case, `ghostty` is launched by i3, such that there is no terminal available.

**Comment by 00-kat** [comment, marked as answer]
You're probably going to have to use the temporary workaround you posted in https://github.com/ghostty-org/ghostty/issues/2435#issuecomment-2564058758 until #2504 is solved.

**Comment by carrascomj** [comment]
The same happens with `open_config`. To note, the `:paste` variant works and the mime type of the temporary file is correct.

Also, as reported before, launching ghostty from another terminal emulator (instead of the window manager) and then executing `write_scrollback_buffer:open` opens the buffer in the parent terminal.
